### PR TITLE
Add OAuth example to sample application

### PR DESCRIPTION
**Pull request by normj**
*Description of changes:*
Add example to sample application how to configure and use Cognito to make authenticated REST API calls.

image

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: samples/Samples/Pages/OAuthExample.cshtml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+<p>
+    Cognito support OAuth 2.0 authorization framework to make authenticated REST API calls. To identify the authenticated user to an
+    API a JSON Web Token (JWT) is generated by the OAuth provider, in this case Cognito, and passed as the <b>Authorization</b> header
+    to the REST API. The format of the value for Authorization header is "<i>bearer &lt;jwt-token&gt;</i>".
+</p>
+
+<h2>Retrieving current user's JWT token.</h2>
+
+<p>
+    In an ASP.NET Core application a JWT token can be retrieved for the current logged in user by injecting ASP.NET Core's UserManager&lt;CognitoUser&gt; service.
```
Comment (ashovlin) [inline samples/Samples/Pages/OAuthExample.cshtml:17]: Bold the class names (UserManager<CognitoUser>, UserManager, CognitoUser) like they are below?

File: samples/Samples/Pages/OAuthExample.cshtml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        this.JwtToken = cognitoUser.SessionTokens.IdToken;
+    }
+}
+</code></pre>
+
+
+@if (Model.IsLoggedIn)
+{
+<p>This is the current logged in user's JWT token:</p>
+<textarea disabled style="width: 100%; box-sizing: border-box;" rows="8">@Model.JwtToken</textarea>
```
Comment (ashovlin) [inline samples/Samples/Pages/OAuthExample.cshtml:40]: This didn't work for me:
image

Running this branch from source, filled out `appsettings.Development.json`.

**Review by ashovlin — COMMENTED** [review]
And general thought - should we throw a markdown copy into the [/docs](https://github.com/aws/aws-aspnet-cognito-identity-provider/tree/master/docs) folder?

It being interactive is cool, but wonder if this is too out of the way?

File: samples/Samples/Pages/OAuthExample.cshtml [hunk 3]
```diff
@@ -0,0 +1,81 @@
+﻿@page
+@model Samples.Pages.OAuthExampleModel
+@{
+}
+
+<h1>OAuth with ASP.NET Core and Amazon Cognito</h1>
+
+<p>
+    Cognito support OAuth 2.0 authorization framework to make authenticated REST API calls. To identify the authenticated user to an
```
[the remark this answers is not in the document]
Comment (normj) [reply samples/Samples/Pages/OAuthExample.cshtml:9]: Fixed

File: samples/Samples/Pages/OAuthExample.cshtml (hunk 1 above)
Comment (normj) [reply samples/Samples/Pages/OAuthExample.cshtml:17]: Done

File: samples/Samples/Pages/OAuthExample.cshtml (hunk 2 above)
Comment (jmsvl) [reply samples/Samples/Pages/OAuthExample.cshtml:40]: It worked for me

image

**Comment by normj** [comment]
Closing for lack of activity

**Pull request closed without merging**